src/test/java/seedu/address/model/AddressBookTest.java

```diff
@@ -78,11 +82,81 @@ public void hasPerson_personWithSameIdentityFieldsInAddressBook_returnsTrue() {
         assertTrue(addressBook.hasPerson(editedAlice));
     }
 
+    @Test
+    public void hasPersonWithSamePhone_personWithSamePhoneInAddressBook_returnsTrue() {
+        addressBook.addPerson(ALICE);
+        Person editedAlice = new PersonBuilder(ALICE).withEmail("alice@example.org").build();
+        assertTrue(addressBook.hasPersonWithSamePhone(editedAlice));
+    }
+
+    @Test
+    public void hasPersonWithSamePhone_personWithDifferentPhoneInAddressBook_returnsFalse() {
+        addressBook.addPerson(ALICE);
+        Person bob = new PersonBuilder().withPhone("11111111").build(); // Use a phone number not used by ALICE
```

Comment:
Perhaps we can abstract these out as static constants to defend against formatting changes for phone in the future? Similar issues elsewhere for other test input parameters